posthog/hogql/transforms/lazy_tables.py

```diff
@@ -21,18 +21,48 @@ def resolve_lazy_tables(
 
 @dataclasses.dataclass
 class JoinToAdd:
-    fields_accessed: Dict[str, List[str]]
+    fields_accessed: Dict[str, List[str | int]]
     lazy_join: LazyJoin
     from_table: str
     to_table: str
 
 
 @dataclasses.dataclass
 class TableToAdd:
-    fields_accessed: Dict[str, List[str]]
+    fields_accessed: Dict[str, List[str | int]]
     lazy_table: LazyTable
 
 
+@dataclasses.dataclass
+class ConstraintOverride:
+    alias: str
+    table_name: str
+    chain_to_replace: List[str | int]
+
+
+class FieldChainReplacer(TraversingVisitor):
+    overrides: List[ConstraintOverride] = {}
+
+    def __init__(self, overrides: List[ConstraintOverride]) -> None:
+        super().__init__()
+        self.overrides = overrides
+
+    def visit_field(self, node: ast.Field):
+        for constraint in self.overrides:
+            if node.chain == constraint.chain_to_replace:
```

Comment:
If this is traversing the entire query again, is it possible that the chain_to_replace matches something unintentionally? It's a super nit edge case but if an outer query reuses an alias/table name that exists on an inner query, it could get overwritten by this override?